backend/compact-connect/lambdas/staff-users/data_model/client.py

```diff
@@ -109,61 +109,116 @@ def update_user_permissions(
         logger.info('Updating staff user permissions', user_id=user_id)
 
         update_expression_parts = {'add': [], 'delete': []}
+
+        # DynamoDB does not support both ADD and DELETE on the same String Set in a single UpdateItem call, so we will
+        # split additions and removals into two calls to prevent a conflict.
```

Comment:
idea being that a delete will undo an add if both are in the string?

Reply:
Strictly speaking, it's not possible for a request to include both the addition and removal because of the structure of the API but, yes, I have the delete after add just so that, if there is ever a conflict, the DELETE will 'win'.